applications/ShapeOptimizationApplication/python_scripts/algorithms/algorithm_relaxed_gradient_projection.py

```diff
@@ -0,0 +1,576 @@
+            gradient_variable = self.constraint_gradient_variables[con_id]["gradient"]
+            mapped_gradient_variable = self.constraint_gradient_variables[con_id]["mapped_gradient"]
+            self.mapper.InverseMap(gradient_variable, mapped_gradient_variable)
+
+        self.inner_iter = 1
+
+        #TODO add convergence criteria, for now only 1 iteration run for testing
+        while not self.__checkInnerConvergence():
+
+            self.direction_has_changed = False
+
+            KM.Logger.PrintInfo("ShapeOpt", "Inner Iteration to Find Shape Update = ", self.inner_iter)
+
+            self.__computeControlPointUpdate()
+
+            self.mapper.Map(KSO.CONTROL_POINT_UPDATE, KSO.SHAPE_UPDATE)
+            self.model_part_controller.DampNodalUpdateVariableIfSpecified(KSO.SHAPE_UPDATE)
+
+            self.__checkConstraintValue()
+            self.inner_iter += 1
+        self.__saveLineSearchData()
+
+
+    # --------------------------------------------------------------------------
+
+    def __checkInnerConvergence(self):
+        KM.Logger.PrintInfo("Check Convergence of the inner loop:")
+        if self.inner_iter == 1:
+            return False
+        elif self.direction_has_changed and self.inner_iter < 100:
+            return False
+        else:
+            return True
+
+    def __checkConstraintValue(self):
+        index = -1
+        for constraint in self.constraints:
+            if self.__isConstraintActive(constraint):
+                index += 1
+                identifier = constraint["identifier"].GetString()
+                constraint_value = self.communicator.getStandardizedValue(identifier)
+                g_a_variable = self.constraint_gradient_variables[identifier]["gradient"]
+                shape_update = KM.Vector()
+                gradient = KM.Vector()
+                self.optimization_utilities.AssembleVector(self.design_surface, gradient, g_a_variable)
+                self.optimization_utilities.AssembleVector(self.design_surface, shape_update, KSO.SHAPE_UPDATE)
+                new_constraint_value = constraint_value + np.dot(gradient, shape_update)
+                KM.Logger.PrintInfo("Constraint ", identifier, "Linearized new value = ", new_constraint_value)
+                if new_constraint_value > 0.0:
+                    if self.relaxation_coefficients[index] < 1.0:
+                        self.relaxation_coefficients[index] = min(self.relaxation_coefficients[index] + 0.02, 1.0)
+                        self.direction_has_changed = True
+                    elif self.correction_coefficients[index] < 2.0:
+                        self.correction_coefficients[index] = min (self.correction_coefficients[index] + 0.02, 2.0)
+                        self.direction_has_changed = True
+                KM.Logger.PrintInfo("Constraint ", identifier, "W_R, W_C = ", self.relaxation_coefficients[index], self.correction_coefficients[index])
```

Comment:
```suggestion
                KM.Logger.PrintInfo("Constraint ", identifier, "w_r, w_c = ", self.relaxation_coefficients[index], self.correction_coefficients[index])
```

doe that make sense?